.ci/magician/github/init.go

```diff
@@ -16,21 +16,18 @@
 package github
 
 import (
-	"fmt"
 	"os"
 )
 
 // Client for GitHub interactions.
 type Client struct {
-	token string
+	classicToken string
+	token        string
 }
 
 func NewClient() *Client {
-	githubToken, ok := os.LookupEnv("GITHUB_TOKEN")
-	if !ok {
-		fmt.Println("Did not provide GITHUB_TOKEN environment variable")
-		os.Exit(1)
+	return &Client{
```

Comment:
Rather than giving the client multiple tokens, we should pass in the token as an argument to `NewClient` - if the caller needs to make calls with different tokens it should create two clients. (It'll also be better to have the GitHub client not have to know about what the env vars are called.)

Having one token per client is a pretty common pattern when making clients like this. Making this change will also make it easier for us to transition to go-github later: https://github.com/google/go-github?tab=readme-ov-file#authentication